site/docs/plugins/conversations.md

```diff
@@ -0,0 +1,900 @@
+This allows you to chat with a single user only in a group chat.
+
+For instance, let's implement the captcha example from up here again, but this time with parallel conversations.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+async function captcha(conversation: MyConversation, ctx: MyContext) {
+  if (ctx.from === undefined) return false;
+  await ctx.reply("Prove you are human! What is the answer to everything?");
+  const { message } = await conversation.waitFrom(ctx.from);
+  return message?.text === "42";
+}
+
+async function enterGroup(conversation: MyConversation, ctx: MyContext) {
+  const ok = await captcha(conversation, ctx);
+
+  if (ok) await ctx.reply("Welcome!");
+  else await ctx.banChatMember();
+}
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
+
+```js
+async function captcha(conversation, ctx) {
+  if (ctx.from === undefined) return false;
+  await ctx.reply("Prove you are human! What is the answer to everything?");
+  const { message } = await conversation.waitFrom(ctx.from);
+  return message?.text === "42";
+}
+
+async function enterGroup(conversation, ctx) {
+  const ok = await captcha(conversation, ctx);
+
+  if (ok) await ctx.reply("Welcome!");
+  else await ctx.banChatMember();
+}
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+Note how we only wait for messages from a particular user.
+
+We can now have a simple handler that enters the conversation when a new member joins.
+
+```ts
+bot.on("chat_member")
+  .filter((ctx) => ctx.chatMember.old_chat_member.status === "left")
+  .filter((ctx) => ctx.chatMember.new_chat_member.status === "member")
+  .use((ctx) => ctx.conversation.enter("enterGroup"));
+```
+
+### Inspecting Active Conversations
+
+You can see how many conversations with which identifier are running.
+
+```ts
+const stats = ctx.conversation.active;
+console.log(stats); // { "enterGroup": 1 }
+```
+
+This will be provided as an object that has the conversation identifiers as keys, and a number indicating the number of running conversations for each identifier.
+
+## How It Works
+
+> [Remember](#three-golden-rules-of-conversations) that the code inside your conversation builder functions must follow three rules.
+> We are now going to see _why_ you need to build them that way.
+
+We are first going to see how this plugin works conceptually, before we elaborate on some details.
+
+### How `wait` Calls Work
+
+Let us switch perspectives for a while, and ask a question from a plugin developer's point of view.
+How to implement a `wait` call in the plugin?
+
+The naïve approach to implementing a `wait` call in the conversations plugin would be to create a new promise, and to wait until the next context object arrives.
+As soon as it does, we resolve the promise, and the conversation can continue.
+
+However, this is a bad idea for several reasons.
+
+**Data Loss.**
+What if your server crashes while waiting for a context object?
```

Comment:
```suggestion

What if your server crashes while waiting for a context object?
```

For me, it's still hard to read. I suggest adding a blank line to make a "subtitle" or make a list

Reply:
Is it still relevant?